core/src/store/function.rs

```diff
@@ -0,0 +1,32 @@
+use crate::data::CustomFunction;
+use crate::result::{Error, Result};
+use async_trait::async_trait;
+
+#[async_trait(?Send)]
+pub trait Function {
```

Comment:
It would be good if we can use name as short as possible but `Function` looks that it has too broad meaning.
In SQL, we are already familiar with `CustomFunction` keyword so how about simply to use `pub trait CustomFunction` and `pub trait CustomFunctionMut` for store traits?